lib/src/lib/dataset.ts

```diff
@@ -0,0 +1,170 @@
+      if (error) {
+        throw error;
+      } else if (status != 0) {
+        throw new Error(`Unzipping failed with exit status ${ status }, stderr: \n${stderr}`);
+      }
+      const infoPath = path.join(tmpDir, "info.csv");
+      if (await fs.access(infoPath, constants.R_OK).then(() => true).catch(() => false)) {
+        const { nonIgnoredFiles, ignoredFile } = (await this.fromCSV(infoPath, ignore)).ok();
+        if (nonIgnoredFiles) {
+          return Result.ok({ nonIgnoredFiles, ignoredFile });
+        }
+        else {
+          throw new Error("Failed to process files");
+        }
+      } else {
+        const nonIgnoredFiles = (await this.fromDirectory(tmpDir)).ok();
+        const ignoredFile = undefined;
+        return Result.ok({ nonIgnoredFiles, ignoredFile });
+      }
+    } finally {
+      await fs.rm(tmpDir, { recursive: true });
+    }
+  }
+
+
+  private async fromCSV(
+    infoPath: string,
+    ignore?: string
+  ): Promise<Result<{ nonIgnoredFiles: File[], ignoredFile: File | undefined }>> {
+    const dirname = path.dirname(infoPath);
+    try {
+      const files = csvParse((await fs.readFile(infoPath)).toString())
+        .map((row:  DSVRowString) => ({
+          filename: row.filename as string,
+          fullName: row.full_name as string,
+          id: row.id as string,
+          status: row.status as string,
+          submissionID: row.submission_id as string,
+          nameEN: row.name_en as string,
+          nameNL: row.name_nl as string,
+          exerciseID: row.exercise_id as string,
+          createdAt: new Date(row.created_at as string),
+          labels: row.label as string || row.labels as string,
+          ignored: row.ignored as string
+        }))
+        .map((row: ExtraInfo) => readPath(path.join(dirname, row.filename), row));
+      const resolvedFiles = await Result.all(files);
+      const ignoredFile = await this.setIgnoredFile(resolvedFiles.ok(), ignore);
+      const nonIgnoredFiles = resolvedFiles.ok().filter(file => file.extra?.ignored !== "true");
+      return Result.ok({ nonIgnoredFiles, ignoredFile });
+    } catch(e) {
+      throw new Error("The given '.csv'-file could not be opened");
+    }
+  }
+
+
+  public async createNameCandidate(paths: string[]): Promise<string> {
+    let nameCandidate = "undefined";
+    if (paths.length === 1) {
+      if (paths[0].toLowerCase().endsWith(".zip")) {
+        nameCandidate = path.basename(paths[0], ".zip");
+      } else if (paths[0].toLowerCase().endsWith(".csv")) {
+        if (paths[0].endsWith("info.csv")) {
+          nameCandidate = path.dirname(paths[0]).split(path.sep).pop() || "undefined";
+        }
+      }
+    }
+    else if (paths.length === 2) {
+      nameCandidate = path.basename(paths[0]) + " & " + path.basename(paths[1]);
+    }
+    return nameCandidate;
+  }
+
+  public async prepareDataset(paths: string[], ignore?: string): Promise<Dataset> {
```

Comment:
The `async` methods create a bit of an awkward setup where you need to construct an empty `Dataset` object first and then call this method to initialize it. I see there is indeed no way to put this part of the code in the constructor because it is async.

I suggest transforming this method to a _factory_ function like this:

```ts

class Dataset {

	constructor(
		public name: string,
		public files: File[],
		public ignore?: File) {
    }

	public static async create(paths: string[], ignore?: string): Promise<Dataset> {
    	// do async stuff
    }
}
```

Using this factory function will then be as ergonomic as:

```ts
const dataset = await Dataset.create(paths, ignore);
```